### PR TITLE
Score endpoints

### PR DESCRIPTION
**Pull request by moui72**
Some file refactors
Significant testing fixture refactor
Testing file refactor

put score endpoint implemented with testing

todo implement score mechanic (non-endpoint) test

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,10 @@
 .DS_Store
-**/*/__pycache__
+**/__pycache__
```
Comment (kurtiss) [inline .gitignore:2]: Was I wrong about this?

File: bgt/datatypes.py [hunk 2]
```diff
@@ -81,6 +81,11 @@ def __hash__(self) -> int:
     def __eq__(self, other) -> bool:
         return str(self) == str(other)
 
+    @classmethod
+    def from_str(cls, qid_str: str) -> QuestionID:
+        temp = qid_str.split("-")
```
Comment (kurtiss) [inline bgt/datatypes.py]: When I have a small number of named arguments, I typically prefer a destructuring assignment in these cases.

File: bgt/game_mechanics.py [hunk 3]
```diff
@@ -79,6 +79,8 @@ class Feedback(BaseModel):
     is_correct: bool = None
     response_text: str = None
 
+    # question: does it make sense for the following to be @validators?
+    # maybe they should be @properties
```
Comment (kurtiss) [inline bgt/game_mechanics.py]: Based on a quick look at the pydantic documentation, it looks like @pydantic.property would make sense.

File: .gitignore (hunk 1 above)
Comment (moui72) [reply .gitignore:2]: I'm not sure, I was having trouble with `tests/pycache` not being ignored, but maybe I made a mistake, I'll have to look at it again

**Pull request merged**